crates/re_sdk/src/binary_stream_sink.rs

```diff
@@ -0,0 +1,200 @@
+use std::sync::mpsc::{Receiver, Sender, SyncSender};
+use std::sync::Arc;
+
+use parking_lot::Mutex;
+
+use re_log_types::LogMsg;
+
+use crate::sink::LogSink;
+use crate::RecordingStream;
+
+/// Errors that can occur when creating a [`BinaryStreamSink`].
+#[derive(thiserror::Error, Debug)]
+pub enum BinaryStreamSinkError {
+    /// Error spawning the writer thread.
+    #[error("Failed to spawn thread: {0}")]
+    SpawnThread(std::io::Error),
+
+    /// Error encoding a log message.
+    #[error("Failed to encode LogMsg: {0}")]
+    LogMsgEncode(#[from] re_log_encoding::encoder::EncodeError),
+}
+
+enum Command {
+    Send(LogMsg),
+    Flush(SyncSender<()>),
+}
+
+impl Command {
+    fn flush() -> (Self, Receiver<()>) {
+        let (tx, rx) = std::sync::mpsc::sync_channel(0); // oneshot
+        (Self::Flush(tx), rx)
+    }
+}
```

Comment:
nit: I found 3 implementations of this, two of them using crossbeam one using mpsc like you here. I like that you bring balance to the world ;)

Reply:
Good enough for FileSink, good enough for me :shrug: